st2common/st2common/service_setup.py

```diff
@@ -170,9 +180,57 @@ def setup(service, config, setup_db=True, register_mq_exchanges=True,
 
     metrics_initialize()
 
+    # Register service in the service registry
+    if service_registry:
+        # NOTE: It's important that we pass start_heart=True to start the hearbeat process
+        register_service_in_service_registry(service=service, capabilities=capabilities,
+                                             start_heart=True)
+
 
 def teardown():
     """
     Common teardown function.
     """
+    # 1. Tear down the database
     db_teardown()
+
+    # 2. Tear down the coordinator
+    coordinator = coordination.get_coordinator()
+    coordination.coordinator_teardown(coordinator)
+
+
+def register_service_in_service_registry(service, capabilities=None, start_heart=True):
+    """
+    Register provided service in the service registry and start the heartbeat process.
+
+    :param service: Service name which will also be used for a group name (e.g. "api").
+    :type service: ``str``
+
+    :param capabilities: Optional metadata associated with the service.
+    :type capabilities: ``dict``
+    """
+    # NOTE: It's important that we pass start_heart=True to start the hearbeat process
+    coordinator = coordination.get_coordinator(start_heart=start_heart)
+
+    member_id = coordination.get_member_id()
+
+    # 1. Create a group with the name of the service
+    if not isinstance(service, six.binary_type):
+        group_id = service.encode('utf-8')
+    else:
+        group_id = service
+
+    try:
+        coordinator.create_group(group_id).get()
+    except GroupAlreadyExist:
+        pass
+
+    # Include common capabilities such as hostname and process ID
+    proc_info = system_info.get_process_info()
+    capabilities['hostname'] = proc_info['hostname']
+    capabilities['pid'] = proc_info['pid']
```

Comment:
Doesn't get_member_id already included this info?  Maybe doesn't matter much, but seems like we're calling system_info.get_process_info twice.

Reply:
Nope, this function doesn't include member info. Just hostname and pid.